docker/beam-environment/entrypoint.sh

```diff
@@ -0,0 +1,362 @@
+    echo " "
+  fi
+}
+
+function send_json_to_spreadsheet() {
+  if [[ -z "$SIMULATIONS_SPREADSHEET_UPDATE_URL" ]]; then
+    echo "WARNING: Can't send updates to the spreadsheet, SIMULATIONS_SPREADSHEET_UPDATE_URL is not set!"
+  else
+    json_data="{\"command\":\"add\",\"type\":\"beam\",\"run\":{ $1 } }"
+    printf "\nSending the following json to the spreadsheet:"
+    echo "$json_data"
+    echo " "
+    curl -X POST -H 'Content-type:application/json' --data "$json_data" "$SIMULATIONS_SPREADSHEET_UPDATE_URL"
+    echo " "
+  fi
+}
+
+
+##
+## logging CPU | RAM usage during simulation
+##
+CPU_RAM_LOG="/app/sources/cpu_ram_usage.csv"
+/app/write_cpu_ram_usage.sh > "$CPU_RAM_LOG" &
+
+
+##
+## location to project folder
+##
+PATH_TO_PROJECT_PARENT="/app/sources"
+cd "$PATH_TO_PROJECT_PARENT" || echo "ERROR: The path '$PATH_TO_PROJECT_PARENT' is not available"
+
+
+##
+## either pull the code or use the code from mounted folder
+##
+if [ "$PULL_CODE" = true ]; then
+  BEAM_NAME="beam"
+  if [ -z "$BEAM_CODE_URL" ]; then BEAM_CODE_URL="https://github.com/LBNL-UCB-STI/beam.git"; fi
+  echo "Pulling the code from github (PULL_CODE set to '$PULL_CODE'), cloning '$BEAM_CODE_URL' into $(pwd)/$BEAM_NAME"
+  git clone --single-branch --branch "$BEAM_BRANCH_NAME" "$BEAM_CODE_URL" "$BEAM_NAME"
+  cd "$BEAM_NAME" || echo "ERROR: The dir '$BEAM_NAME' is not available"
+
+  if [[ $BEAM_COMMIT_SHA ]]; then
+    echo "Resetting the branch '$BEAM_BRANCH_NAME' to commit '$BEAM_COMMIT_SHA'"
+    git reset --hard "$BEAM_COMMIT_SHA";
+  else
+    BEAM_COMMIT_SHA=$(git log -1 --pretty=format:%H)
+    echo "Using the latest commit in the branch '$BEAM_BRANCH_NAME' - '$BEAM_COMMIT_SHA'"
+  fi
+else
+  echo "Pulling the code is disabled (PULL_CODE set to '$PULL_CODE'), using code mounted to '$PATH_TO_PROJECT_PARENT'."
+  BEAM_COMMIT_SHA=$(git log -1 --pretty=format:%H)
+  BEAM_BRANCH_NAME=$(git symbolic-ref --short HEAD)
+  echo "Using the latest commit in the branch '$BEAM_BRANCH_NAME' - '$BEAM_COMMIT_SHA'"
+fi
+
+
+##
+## remember the BEAM working folder location
+##
+BEAM_PATH=$(pwd)
+echo "Working from '$BEAM_PATH'"
+
+
+##
+## Required to avoid configuring ssh keys because by-default all our git data submodules configured to use ssh.
+##
+if [ "$ENFORCE_HTTPS_FOR_DATA_REPOSITORY" = true ]; then
+  echo "Forcing git to use https over ssh\git url. In order to pull public repository without configuring ssh keys."
+  git config --global url."https://github.com/LBNL-UCB-STI".insteadOf "git@github.com:LBNL-UCB-STI"
+fi
+
+
+##
+## pulling data from github if enabled or checking if data location was mounted separately
+##
+if [ "$PULL_DATA" = true ]; then
+  echo "Pulling the data from github (PULL_DATA set to '$PULL_DATA')."
+  production_data_submodules=$(git submodule | awk '{ print $2 }')
+  for i in $production_data_submodules;
+    do
+      if [[ $BEAM_CONFIG  == $i* ]];
```

Comment:
It is a check to find if beam config contains a part of production module paths. It seems double sided wildcard does not required because we want to check if config starts from the same path, not just contains it somewhere in the middle

Reply:
If that is the spec then sounds good to me